packages/ckeditor5-image/src/image/utils.js

```diff
@@ -59,6 +59,26 @@ export function getSelectedImageWidget( selection ) {
 	return null;
 }
 
+/**
+ * Returns a image widget editing view element if one is among the selection's ancestors.
+ *
+ * @param {module:engine/view/selection~Selection|module:engine/view/documentselection~DocumentSelection} selection
+ * @returns {module:engine/view/element~Element|null}
+ */
+export function getImageWidgetAncestor( selection ) {
+	let parent = selection.getFirstPosition().parent;
+
+	while ( parent ) {
+		if ( parent.name === 'figure' && parent.getCustomProperty( 'image' ) ) {
```

Comment:
Let's not re-invent the wheel :)

```suggestion
		if ( parent.is( 'element' ) && isImageWidget( parent ) ) {
```

Alternatively `isImageWidget()` starts accepting view node (instead of element) and `parent.is( 'element' )` goes there (+tests).